languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -10069,6 +10069,107 @@ though, and has since been slightly modified)-->
+                  <token postag="D f p|D e p" postag_regexp="yes"/>
+                  <token postag="N f p"/>
+              </pattern>
+              <message>Une erreur semble être détectée dans cette phrase. Souhaitez-vous écrire <suggestion>quelles</suggestion> ?</message>
+              <example correction="quelles">Il le fera <marker>qu'elles</marker> qu'en soient les conséquences.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <token postag="N f p" skip="-1"/>
+                  <marker>
+                      <token regexp="yes">quelle|quel|quels|qu(&apostrophe;)</token>
+                      <token regexp="yes" min="0">elle|elles</token>
+                  </marker>
+                  <token regexp="yes">qu(&apostrophe;)</token>
+                  <token>elles</token>
+              </pattern>
+              <message>Une erreur semble être détectée dans cette phrase. Souhaitez-vous écrire <suggestion>quelles</suggestion> ?</message>
+              <example correction="quelles">On trouvera des solutions, <marker>qu'elles</marker> qu'elles soient.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <token>quelle</token>
+                  <marker>
+                      <token>quelle</token>
+                  </marker>
+                  <token>soit</token>
+              </pattern>
+              <message>Une erreur de pronom semble être détectée. Souhaitez-vous écrire <suggestion>qu'elle</suggestion> ?</message>
+              <example correction="qu'elle">Donne-moi ta carte, quelle <marker>quelle</marker> soit.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">qu(&apostrophe;)</token>
+                      <token>elle</token>
+                      <token regexp="yes">qu(&apostrophe;)</token>
+                  </marker>
+                  <token>un</token>
+              </pattern>
+              <message>Une erreur semble être détectée dans cette phrase. Souhaitez-vous écrire <suggestion>quelqu'un</suggestion> ?</message>
+              <example correction="quelqu'un">Est-ce que <marker>qu'elle qu'</marker>un peut m'aider.</example>
+              <example correction="quelqu'un">Y-a-t-il <marker>qu'elle qu'</marker>un ici ?</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <token>quelles</token>
+                  <marker>
+                      <token>quelles</token>
+                  </marker>
+                  <token>soient</token>
+              </pattern>
+              <message>Une erreur de pronom semble être détectée. Souhaitez-vous écrire <suggestion>qu'elles</suggestion> ?</message>
+              <suggestion>qu'elles</suggestion>
+              <example correction="qu'elles">Donne-moi tes billes, quelles <marker>quelles</marker> soient.</example>
+          </rule>
+      </rulegroup>
```

Comment:
Please check the rule editor results as well as the regex in all the subrules and let me know in a comment here when the rule is ready for reviewing.

Reply:
I have made all the changes required so the rule is ready .